### PR TITLE
Add OpenAI-compatible EvalRunner for vLLM/KServe endpoints

### PR DESCRIPTION
**Pull request by Sridhar1030**
## Summary
- Adds an `openai-compatible` runner that evaluates models served via standard OpenAI-compatible `/v1/chat/completions` APIs
- Enables the harness to evaluate models deployed on vLLM, KServe, Ollama, LiteLLM, or any OpenAI-compatible endpoint
- No new dependencies -- uses stdlib `urllib.request`
- Registers in `RUNNERS` dict so `runner.type: openai-compatible` works in eval.yaml

## Motivation
The harness currently only supports Claude Code as a runner. This adds support for evaluating any model behind an OpenAI-compatible API, making the harness useful for a much wider range of deployment targets (vLLM, KServe, Ollama, LiteLLM, etc.).

## Usage
```yaml
# eval.yaml
runner:
  type: openai-compatible
```

Set `OPENAI_BASE_URL` and optionally `OPENAI_API_KEY` in the environment, or pass `base_url`/`api_key` directly.

## Test plan
- 58 unit tests covering constructor validation, request format, success/error paths, ABC compliance, harness integration, and edge cases (null content, empty choices, unicode, large responses)
- Tests use a real local HTTP mock server (not mocked at function level)
- All 111 tests in the repo pass (58 new + 53 existing)

## Release Notes

* **New Features**
  * Introduced support for OpenAI-compatible evaluation runners, enabling evaluation against custom API endpoints with configurable models and authentication.
  * Added token usage tracking, structured error handling, and comprehensive artifact logging of evaluation results.

### CODE REVIEW

File: agent_eval/agent/openai_compatible.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+from typing import Optional
+from urllib.parse import urlparse
+
+from .base import EvalRunner, RunResult
+
+
+_MAX_RESPONSE_BYTES = 50 * 1024 * 1024  # 50 MB hard cap
+
+
+class OpenAICompatibleRunner(EvalRunner):
```
Comment (astefanutti) [inline agent_eval/agent/openai_compatible.py:24]: I wonder whether this could become a Responses API compatible runner so it can execute the skill, like:
https://openai.com/index/equip-responses-api-computer-environment/#agent-skills

I've created #49 which outlines what that could look like.

Comment (Sridhar1030) [reply agent_eval/agent/openai_compatible.py:24]: Looked at the PR, 
the Responses API runner with Skills + Shell would be the proper equivalent to ClaudeCodeRunner for agentic skill execution. 
This PR covers the simpler completions use case (scoring raw model output from vLLM/KServe without needing an agentic loop). Happy to keep this as-is for the completions use case and build on top of #49 once it lands.

**Comment by astefanutti** [comment]
@Sridhar1030 I think this has been superseded by #65 and can be closed right?

**Comment by Sridhar1030** [comment]
Yes , closing it
